addons/mail_plugin/tests/common.py

```diff
@@ -2,13 +2,35 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import json
+from contextlib import contextmanager
 from unittest.mock import patch
 
 from odoo.addons.mail.tests.common import mail_new_test_user
 from odoo.http import request
 from odoo.tests.common import HttpCase
 
 
+def mock_auth_method(login):
+    """Mock the Outlook auth method.
+
+    This must be used as a method decorator.
+
+    :param login: Login of the user used for the authentication
+    """
+    def patched_auth_method_outlook(*args, **kwargs):
+        request.uid = request.env['res.users'].search([('login', '=', login)], limit=1).id
+
+    def f(test_func):
+        def mocked_test_func(*args, **kwargs):
+            with patch(
+                    'odoo.addons.mail_plugin.models.ir_http.IrHttp'
+                    '._auth_method_outlook',
+                    new=patched_auth_method_outlook):
+                return test_func(*args, **kwargs)
+        return mocked_test_func
+    return f
```

Comment:
Sounds a bit complicated.
Can't we use a "yield" to make it simpler?

It's what I did here for example:
https://github.com/odoo/enterprise/blob/master/social/tests/tools.py#L57

Maybe it doesn't apply in this case?

Reply:
(Also a good reference to check: `def users` that we already use with `@users()` in tests).

Reply:
I prefer your version with @contextlib and yield :) done